src/components/search.js

```diff
@@ -1,15 +1,47 @@
 import React, { Component } from 'react';
+import SearchWidget from './searchWidget';
+import $ from 'jquery';
 
 
 export default class Search extends Component {
+    constructor(props){
+        super(props);
+        this.state={
+            inputValue: '',
+            keyword: ''
+        };
+    }
+
+    componentWillMount(){
```

Comment:
same thing as below. should we avoid componentWillMount?

Reply:
This has been changed to DidMount